Docs/source/developers/testing.rst

```diff
@@ -47,97 +30,149 @@ You can run those locally before you push to save some time, install them once l
+* Add the **restart test** ``test_3d_laser_acceleration_restart``, with input file in ``Examples/Tests/laser_acceleration/`` and default restart analysis script provided by WarpX:
+
+  .. code-block:: sh
 
-    $ ls ./test_dir/rt-WarpX/WarpX-tests/2021-04-30/pml_x_yee/
-    analysis_pml_yee.py     # Python analysis script
-    inputs_2d               # input file
-    main2d.gnu.TEST.TPROF.MTMPI.OMP.QED.ex  # executable
-    pml_x_yee.analysis.out  # Python analysis output
-    pml_x_yee.err.out       # error output
-    pml_x_yee.make.out      # build output
-    pml_x_yee_plt00000/     # data output (initialization)
-    pml_x_yee_plt00300/     # data output (last time step)
-    pml_x_yee.run.out       # test output
+       add_warpx_test(
+           test_3d_laser_acceleration_restart # name
+           3 # dims
+           2 # nprocs
+           OFF # eb
+           Examples/Tests/laser_acceleration/inputs_test_3d_laser_acceleration_restart # inputs
+           Examples/analysis_default_restart.py # analysis
+           diags/diag1000100 # output (plotfile)
+           test_3d_laser_acceleration # dependency
+       )
 
+  Note that the restart has an explicit dependency, namely it can run only provided that the original test, from which the restart checkpoint files will be read, runs first.
 
-Add a test to the suite
------------------------
+* A more complex example. Add the **PICMI test** ``test_rz_laser_acceleration_picmi``, with custom command-line arguments ``--test`` and ``dir``, input file and analysis script in ``Examples/Tests/laser_acceleration/``, and openPMD time series output:
 
-There are three steps to follow to add a new automated test (illustrated here for PML boundary conditions):
+  .. code-block:: sh
 
-* An input file for your test, in folder `Example/Tests/...`. For the PML test, the input file is at ``Examples/Tests/pml/inputs_2d``. You can also re-use an existing input file (even better!) and pass specific parameters at runtime (see below).
-* A Python script that reads simulation output and tests correctness versus theory or calibrated results. For the PML test, see ``Examples/Tests/pml/analysis_pml_yee.py``. It typically ends with Python statement ``assert( error<0.01 )``.
-* If you need a new Python package dependency for testing, add it in ``Regression/requirements.txt``
-* Add an entry to ``Regression/WarpX-tests.ini``, so that a WarpX simulation runs your test in the continuous integration process, and the Python script is executed to assess the correctness. For the PML test, the entry is
+       add_warpx_test(
+           test_rz_laser_acceleration_picmi # name
+           RZ # dims
+           2  # nprocs
+           OFF # eb
+           "Examples/Tests/laser_acceleration/inputs_test_rz_laser_acceleration_picmi.py --test --dir 1" # inputs
+           Examples/Tests/laser_acceleration/analysis.py # analysis
+           diags/diag1/ # output (openPMD time series)
+           OFF # dependency
```

Comment:
```suggestion
           test_rz_laser_acceleration_picmi  # name
           RZ  # dims
           2   # nprocs
           OFF  # eb
           "Examples/Tests/laser_acceleration/inputs_test_rz_laser_acceleration_picmi.py --test --dir 1"  # inputs
           Examples/Tests/laser_acceleration/analysis.py  # analysis
           diags/diag1/  # output (openPMD time series)
           OFF  # dependency
```

Reply:
Done in [8e64870](https://github.com/ECP-WarpX/WarpX/pull/5068/commits/8e64870fe4fe42fd6c88f69741fe7761d30226c6).